### ISSUE TITLE
Add a generic annotations pattern

### ISSUE THREAD

**Issue by cmungall** [biolink/biolinkml#167]
We often want to add annotated annotations

e.g.

```
classes:
  my class:
    annotations:
      altLabel:
         - 
           value: some other label
           annotations:
             source:
               value: PMID:12345
```

In principle this allows unb ounded nesting as in owl2 (though I have never needed >1 level)

this would also allow people to use AP/metadata slots without to describe their model wiithout advance inclusion in metamodel: e.g. #150

**Comment by cmungall** [comment]
the concern with the above is that we have undefined keys

an alternative would be:

```
classes:
  my class:
    annotations:
      property: altLabel:
      value: some other label
      annotations:
           source:
             value: PMID:12345
```

**Comment by nlharris** [comment]
In biolink/biolink-model#440, Chris said "I think this ticket should be closed: biolink/biolinkml#167"

**Comment by hsolbrig** [comment]
Done
